src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Telemetry/VsTelemetryService.cs

```diff
@@ -0,0 +1,238 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.Telemetry;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Telemetry
+{
+    [Export(typeof(ITelemetryService))]
+    [SuppressMessage("Microsoft.Cryptographic.Standard", "CA5350:MD5CannotBeUsed", Justification = "MD5 hash is not being used for security")]
+    internal class VsTelemetryService : ITelemetryService
+    {
+        /// <summary>
+        /// Prefix for events.
+        /// </summary>
+        private const string EventPrefix = "vs/projectsystem/managed/";
```

Comment:
one of reason you want to talk to them is to see whether you want to include your data to office cooked project detail stream.